src/imex.rs

```diff
@@ -295,59 +295,81 @@ pub(crate) async fn import_backup_stream<R: tokio::io::AsyncRead + Unpin>(
     let mut archive = Archive::new(backup_file);
 
     let mut entries = archive.entries()?;
-
+    let mut blobs = Vec::new();
     // We already emitted ImexProgress(10) above
     let mut last_progress = 10;
+    const PROGRESS_MIGRATIONS: u64 = 999;
     let mut total_size = 0;
-    while let Some(mut f) = entries
-        .try_next()
-        .await
-        .context("Failed to get next entry")?
-    {
-        total_size += f.header().entry_size()?;
-        let progress = std::cmp::min(
-            1000 * total_size.checked_div(file_size).unwrap_or_default(),
-            999,
-        );
+    let mut res: Result<()> = loop {
+        let mut f = match entries.try_next().await {
+            Ok(Some(f)) => f,
+            Ok(None) => break Ok(()),
+            Err(e) => break Err(e).context("Failed to get next entry"),
+        };
+        total_size += match f.header().entry_size() {
+            Ok(size) => size,
+            Err(e) => break Err(e).context("Failed to get entry size"),
+        };
+        let max = PROGRESS_MIGRATIONS - 1;
+        let progress = std::cmp::min(total_size / (file_size / max + 1), max);
```

Comment:
Fixed the progress calculation, before `total_size.checked_div(file_size)` was giving 0 if `total_size < file_size`. Now it's not accurate for small `file_size`s, but i think it's not important as the import progress is very fast in this case.

Reply:
Moved to a separate commit